core/lib/multivm/src/versions/vm_latest/utils/l2_blocks.rs

```diff
@@ -22,13 +22,19 @@ pub(crate) fn get_l2_block_hash_key(block_number: u32) -> StorageKey {
     )
 }
 
-pub(crate) fn assert_next_block(prev_block: &L2Block, next_block: &L2BlockEnv) {
+pub(crate) fn assert_next_block(
+    prev_block: &L2Block,
+    next_block: &L2BlockEnv,
+    protocol_version: ProtocolVersionId,
+) {
     if prev_block.number == 0 {
         // Special case for the first block it can have the same timestamp as the previous block.
         assert!(prev_block.timestamp <= next_block.timestamp);
     } else {
         assert_eq!(prev_block.number + 1, next_block.number);
-        assert!(prev_block.timestamp < next_block.timestamp);
+        if protocol_version.is_pre_fast_blocks() {
```

Comment:
Dumb question: Will timestamps be no longer monotonic in the new protocol version? Or would they be non-strictly monotonic (`prev_block.timestamp <= next_block.timestamp`)?

Reply:
They would be non-strictly monotonic. I see, it makes sense to assert it as well